### PR TITLE
radare2: Try to detect color support and set scr.color accordingly.

### PR DESCRIPTION
**Pull request by vifino**
Hi.

This PR tries to set `scr.color` to the correct setting for the current terminal.
It does not touch `scr.color` when it's not known what depth the terminal actually supports.

The setting should be overwritable by the RC.

### CODE REVIEW

**Comment by radare** [comment]
Then its not fine

**Comment by ret2libc** [comment]
I think it messes up with my colors because we don't have a good enough default theme for `scr.color=3`

**Comment by ret2libc** [comment]
And probably because I already have colorized color scheme on my terminal :) but the change looks good.

File: binr/radare2/radare2.c [hunk 1]
```diff
@@ -527,6 +527,22 @@ int main(int argc, char **argv, char **envp) {
 		return 0;
 	}
 
+	// Try to set the correct scr.color for the current terminal.
```
Comment (radare) [inline binr/radare2/radare2.c]: move all this logic into a separate function

Comment (vifino) [reply binr/radare2/radare2.c]: okay, assuming in the same file?

File: binr/radare2/radare2.c [hunk 2]
```diff
@@ -527,6 +527,22 @@ int main(int argc, char **argv, char **envp) {
...
+	if ((tmp = r_sys_getenv ("COLORTERM")) && (r_str_endswith (tmp, "truecolor") || r_str_endswith (tmp, "24bit"))) {
```
Comment (radare) [inline binr/radare2/radare2.c]: What happens if the user specifies a different value in radare2rc or the -e flags?

Comment (ret2libc) [reply binr/radare2/radare2.c]: from my tests, it works as you'd expect. radare2rc/-e have precedence over this automagic setting.

Comment (vifino) [reply binr/radare2/radare2.c]: yeah, that was the idea behind it. something that selects a sane default for the current console.

File: binr/radare2/radare2.c (hunk 1 above)
Comment (vifino) [reply binr/radare2/radare2.c]: hmm, somehow my compiler seems to throw away the function and pretend it doesn't exist, even though it's defined right above the main function.

Comment (ret2libc) [reply binr/radare2/radare2.c]: can you push it so we can have a look?

Comment (vifino) [reply binr/radare2/radare2.c]: will do

**Comment by vifino** [comment]
pushed the fix

**Comment by radare** [comment]
You won an r2coin!!1

**Comment by vifino** [comment]
!!!!!

**Pull request merged**